### PR TITLE
examples/fluids: Get Viscous flux from solution for flat plate boundary layer problems

### PR DESCRIPTION
**Pull request by jrwrigh**
Use solution gradients to calculate viscous flux in the boundary integral terms.

TODOs:
- [x] Add P_0 to Newtonian context
    - [x] Set new newtonian_ctx P0 in blasius.c
- [x] Move STGStrongInflow QF into newtonian.h
- [x] Move Blasius_Outflow QF into newtonian.h
- [x] Add viscous flux to the inflow and outflow QFs
- [x] Set the inflow and outflow QFs in newtonian.c
- [x] Move State structs and associated helper functions to separate header file
- [x] Pass solution gradient to inflow and outflow QFs
    - [x] Make name `Grad q`
- [x] Make new tests (accounting for change in boundary conditions)

### CODE REVIEW

File: examples/fluids/qfunctions/newtonian.h
Comment (jedbrown) [inline examples/fluids/qfunctions/newtonian.h:18]: This commit should delete the now-duplicate code from this file. (It doesn't even compile due to redefinition errors.)

Comment (jrwrigh) [reply examples/fluids/qfunctions/newtonian.h:18]: Yep, I realized that soon after I pushed, but forgot to push the fix up last night. Somehow got dropped after moving my local branch from a custom `jed/fluids-jacobian`+`main` branch to `jed/fluid-jacobian` after you merged in `main`.

**Comment by jrwrigh** [comment]
Assuming tests pass, this should be ready for review/merge.

File: examples/fluids/qfunctions/stg_shur14.h
Comment (jedbrown) [inline examples/fluids/qfunctions/stg_shur14.h:378]: @LeilaGhaffari Heads up about this incoming code. If we use `FluxInviscid_fwd`, it'll help make it match what you're doing with primitive variables.

File: examples/fluids/src/setuplibceed.c
Comment (jedbrown) [inline examples/fluids/src/setuplibceed.c:286]: We can get rid of this now, right?

File: examples/fluids/src/setuplibceed.c
Comment (jedbrown) [inline examples/fluids/src/setuplibceed.c:654]: And this?

File: examples/fluids/src/setuplibceed.c
Comment (jedbrown) [inline examples/fluids/src/setuplibceed.c:700]: +

**Review by jedbrown — APPROVED** [review]
Generally looks good, though I think we should remove the `"dummy"` parts (with https://github.com/CEED/libCEED/pull/988, which should merge momentarily). You could cherry-pick 84e7b512ee4f3ef84f7ca44736ead8e0ffaedab4 to do that.

File: examples/fluids/src/setuplibceed.c
Comment (jedbrown) [reply examples/fluids/src/setuplibceed.c:286]: #988 is merged so I'd suggest just cherry-picking the above. Are you planning on a squash-merge or doing some light squashing in this branch's history?

**Comment by jedbrown** [comment]
Do you want this squash-merged or normal merged?

**Comment by jrwrigh** [comment]
Sorry I missed this. I'm ok with either. I can do a quick bit of history clean up if we want it normal merged.

Also, am I "allowed" to merge it in myself? I know I have write access, but still a bit apprehensive of doing it. It's more power than I'm used to wielding. haha

**Pull request merged**
